### PR TITLE
test(debt_payments): cover validation and wire

### LINKED ISSUE #603: test(debt_payments): add unit tests for internal/debt_payments
## Motivation

`internal/debt_payments` has zero test files. Liability payment tracking affects net worth calculations — regressions corrupt headline numbers.

## Implementation information

- Add tests for payment store CRUD operations
- Cover edge cases: zero payments, multi-currency (when added), payment date validation
- Use testcontainers-go pattern consistent with rest of backend-go

## Supporting documentation

Part of umbrella #598

### PR DESCRIPTION
**Pull request by Automaat**
## Motivation

`internal/debt_payments` had zero test files. Liability payment tracking flows into net worth — regressions corrupt headline numbers. Closes #603.

## Implementation information

New `handler_test.go` covers the pure surface area of the package (no DB needed; store CRUD is exercised by bb-tests):

- `isNull` against `null`, whitespace-wrapped `null`, and false positives like `"null"` / `0`
- `requireAmount` (renamed from `requirePositiveDecimal` — only ever called for "amount"): missing, null, garbage, zero, negative, positive int, positive decimal
- `requireOwnerUserID` (renamed from `requireIntOrNull`): missing, explicit null → nil, non-int, integer
- `requireDateNotFuture`: missing, wrong format, future rejected, today accepted, past accepted
- `buildCreateRequest`: missing amount, amount-fails-first ordering, full valid request, null owner allowed
- `toResponse`: identity fields preserved, decimal → pyFloat conversion, owner pointer preserved
- `isoDate` marshals as `"YYYY-MM-DD"`
- `isoNaive` marshals without timezone marker (matches Python's naive datetime)
- `pyFloat` marshals integers as `100.0` and preserves decimals

The three single-use validation helpers were renamed and had their unused `key`/`msg` parameters inlined as `const` — the `unparam` linter flagged that they only ever take one value each, and removing the params is the simpler fix than carrying dead parameters.

## Supporting documentation

Part of umbrella #598.

> Changelog: test(debt_payments): cover validation and wire

### CODE REVIEW

File: backend-go/internal/debt_payments/handler_test.go [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+		if vErr == nil || vErr.Msg != "must be YYYY-MM-DD" {
+			t.Errorf("want must be YYYY-MM-DD, got %+v", vErr)
+		}
+	})
+	t.Run("future rejected", func(t *testing.T) {
+		future := time.Now().UTC().AddDate(0, 0, 1).Format("2006-01-02")
+		_, vErr := requireDateNotFuture(map[string]json.RawMessage{"date": rawJSON(`"` + future + `"`)})
+		if vErr == nil || vErr.Msg != "Date cannot be in the future" {
+			t.Errorf("want future-rejection error, got %+v", vErr)
+		}
```
[the remark this answers is not in the document]
Comment (Automaat) [reply backend-go/internal/debt_payments/handler_test.go:137]: Switched to a far-future 3000-01-01 date in a891368.

File: backend-go/internal/debt_payments/handler_test.go [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+	if err != nil {
+		t.Fatalf("marshal: %v", err)
+	}
+	if string(got) != `"2025-01-05"` {
+		t.Errorf("want \"2025-01-05\", got %s", got)
+	}
+}
+
+func TestIsoNaiveMarshalStripsTimezone(t *testing.T) {
+	loc, _ := time.LoadLocation("Europe/Warsaw")
```
[the remark this answers is not in the document]
Comment (Automaat) [reply backend-go/internal/debt_payments/handler_test.go]: Switched to `time.FixedZone("CET", 3600)` in a891368 — removes the tzdata dependency.
